packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -54,27 +54,39 @@ enum AliasState {
 class DataStore implements IDataStore {
     private aliasState: AliasState = AliasState.None;
     private alias: string | undefined;
+    private aliasResult: Promise<AliasResult> | undefined;
 
     async trySetAlias(alias: string): Promise<AliasResult> {
         if (alias.includes("/")) {
             throw new UsageError(`The alias cannot contain slashes: '${alias}'`);
         }
 
         switch (this.aliasState) {
-            // If we're already aliasing, do nothing
-            case AliasState.Aliasing:
-                return "Aliasing";
+            // If we're already aliasing, check if it's for the same value and return
+            // the stored promise, otherwise return 'AlreadyAliased'
+            case AliasState.Aliasing: {
+                assert(this.aliasResult !== undefined, "There should be a cached promise of in-progress aliasing");
+                const pendingAliasResult = await this.aliasResult;
+                return this.alias === alias ? pendingAliasResult : "AlreadyAliased";
```

Comment:
I feel like it should simply fall through to next case, i.e. return should not depend on pendingAliasResult - if alias is assigned to a name that this request requested, then it's a success.